packages/xrpl/test/integration/transactions/depositPreauth.test.ts

```diff
@@ -32,4 +40,192 @@ describe('DepositPreauth', function () {
+        TransactionType: 'DepositPreauth',
+        Account: testContext.wallet.classicAddress,
+        AuthorizeCredentials: [authorizeCredentialObj],
+      }
+      await testTransaction(
+        testContext.client,
+        authCredDepositPreauthTx,
+        testContext.wallet,
+      )
+
+      const UnauthCredDepositPreauthTx: DepositPreauth = {
+        TransactionType: 'DepositPreauth',
+        Account: testContext.wallet.classicAddress,
+        UnauthorizeCredentials: [authorizeCredentialObj],
+      }
+      await testTransaction(
+        testContext.client,
+        UnauthCredDepositPreauthTx,
+        testContext.wallet,
+      )
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'Invalid multiple of auth and unauth provided',
+    async () => {
+      const subjectWallet = await generateFundedWallet(testContext.client)
+
+      const credentialCreateTx: CredentialCreate = {
+        TransactionType: 'CredentialCreate',
+        Account: testContext.wallet.classicAddress,
+        Subject: subjectWallet.classicAddress,
+        CredentialType: stringToHex('Test Credential Type'),
+      }
+
+      await testTransaction(
+        testContext.client,
+        credentialCreateTx,
+        testContext.wallet,
+      )
+
+      const credentialAcceptTx: CredentialAccept = {
+        TransactionType: 'CredentialAccept',
+        Account: subjectWallet.classicAddress,
+        Issuer: testContext.wallet.classicAddress,
+        CredentialType: stringToHex('Test Credential Type'),
+      }
+
+      await testTransaction(
+        testContext.client,
+        credentialAcceptTx,
+        subjectWallet,
+      )
+
+      const authorizeCredentialObj: AuthorizeCredential = {
+        Credential: {
+          Issuer: testContext.wallet.classicAddress,
+          CredentialType: stringToHex('Test Credential Type'),
+        },
+      }
+
+      const wallet2 = Wallet.generate()
+      await fundAccount(testContext.client, wallet2)
+      const tx: DepositPreauth = {
+        TransactionType: 'DepositPreauth',
+        Account: testContext.wallet.classicAddress,
+        AuthorizeCredentials: [authorizeCredentialObj],
+        UnauthorizeCredentials: [authorizeCredentialObj],
+      }
+      await testTransaction(testContext.client, tx, testContext.wallet)
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'Nonexistent Credential to authorize',
+    async () => {
+      const authorizeCredentialObj: AuthorizeCredential = {
+        Credential: {
+          Issuer: testContext.wallet.classicAddress,
+          CredentialType: stringToHex('Test Credential Type'),
+        },
+      }
+
+      const wallet2 = Wallet.generate()
+      await fundAccount(testContext.client, wallet2)
+      const tx: DepositPreauth = {
+        TransactionType: 'DepositPreauth',
+        Account: testContext.wallet.classicAddress,
+        AuthorizeCredentials: [authorizeCredentialObj],
+      }
+      await testTransaction(testContext.client, tx, testContext.wallet)
```

Comment:
this test-case is vacuous, because it cannot throw a `tec` error despite the non-existence of such credentials.

My understanding is that `DepositPreauth` transaction doesn't sanity-check the AuthorizeCredentials parameters.

Reply:
After reviewing other integration tests, I see we don't have negative case integration tests elsewhere. Have removed this for now, can add later as part of a separate issue

Reply:
I also don't believe it falls under the purview of client libraries to test all the possible edge-cases of a feature. `rippled` already includes such tests.